### PR TITLE
Social login API

### PR DESCRIPTION
**Pull request by UmeshBhosale**

### CODE REVIEW

**Comment by UmeshBhosale** [comment]
@coolbung Please review this MR for social login.

File: easysocial/easysocial.php [hunk 1]
```diff
@@ -34,5 +34,6 @@ public function __construct(&$subject, $config = array())
 		$this->setResourceAccess('terms', 'public', 'post');
 		$this->setResourceAccess('slogin', 'public', 'post');
```
Comment (coolbung) [inline easysocial/easysocial.php]: Are these needed anymore?

File: easysocial/easysocial/sociallogin.php [hunk 2]
```diff
@@ -10,116 +10,757 @@
...
+		this->plugin->setResponse(JText::_('PLG_API_EASYSOCIAL_UNSUPPORTED_METHOD_MESSAGE'));
 	}
 
-	/**	  
+	/**
 	 * POST data
-	 * 	 
-	 * @return  JSON
+	 *
+	 * @return true
```
Comment (coolbung) [inline easysocial/easysocial/sociallogin.php]: What does return true mean ? Does this method return anything at all ?

File: easysocial/easysocial/sociallogin.php [hunk 3]
```diff
@@ -10,116 +10,757 @@
...
-			if (!$jUserEmailId)
+		$app               = JFactory::getApplication();
+		$this->accessToken = $app->input->get('access_token', '', 'STR');
+		$this->provider    = $app->input->get('provider', '', 'STR');
+		$obj               = new stdClass;
+		$this->is_use_jfb = JComponentHelper::isEnabled('com_jfbconnect', true);
+
+		if ($this->accessToken)
+		{
+			$objFbProfileData = $this->jfbGetUser($this->accessToken);
```
Comment (coolbung) [inline easysocial/easysocial/sociallogin.php]: Why isnt there a check to verify if JFB is installed ?

File: easysocial/easysocial/sociallogin.php [hunk 4]
```diff
@@ -10,116 +10,757 @@
...
+	 *
+	 * @param   OBJECT  $accessToken  accessToken
+	 *
+	 * @return true
+	 */
+	public function jfbGetUser($accessToken)
+	{
+		if ($this->provider == 'facebook')
+		{
+				$url          = 'https://graph.facebook.com/v2.2/me';
```
Comment (coolbung) [inline easysocial/easysocial/sociallogin.php]: Joomla already has facebook and google libraries. Use those.

File: easysocial/easysocial/sociallogin.php [hunk 5]
```diff
@@ -10,116 +10,757 @@
...
+						$db->quote(date('Y-m-d H:i:s')),
+						$db->quote(date('Y-m-d H:i:s')),
+						$db->quote('"' . $this->accessToken . '"'),
+						1,
+						$db->quote(json_encode($params)),
+						$db->quote($this->provider)
+		);
+
+		// Prepare the insert query.
+		$query->insert($db->quoteName('#__jfbconnect_user_map'))->columns($db->quoteName($columns))->values(implode(',', $values));
```
Comment (coolbung) [inline easysocial/easysocial/sociallogin.php]: No direct queries to JFB. Use their methods.

**Pull request closed without merging**